clmm/support/mock_data.py

```diff
@@ -404,17 +457,32 @@ def _compute_photoz_pdfs(galaxy_catalog, photoz_sigma_unscaled):
         galaxy_catalog['pzsigma'] * \
         np.random.standard_normal(len(galaxy_catalog))
 
-    pzbins_grid, pzpdf_grid = [], []
-    for row in galaxy_catalog:
-        pdf_range = row['pzsigma']*10.
-        zmin, zmax = row['z']-pdf_range, row['z']+pdf_range
-        zbins = np.arange(zmin, zmax, 0.03)
-        pzbins_grid.append(zbins)
-        pzpdf_grid.append(
-            np.exp(-0.5*((zbins-row['z'])/row['pzsigma'])**2)/np.sqrt(2*np.pi*row['pzsigma']**2))
-    galaxy_catalog['pzbins'] = pzbins_grid
-    galaxy_catalog['pzpdf'] = pzpdf_grid
-
+    if galaxy_catalog.pzpdf_info['type'] is None:
+        pass
+    elif galaxy_catalog.pzpdf_info['type']=='shared_bins':
+        galaxy_catalog.pzpdf_info['zbins'] = np.arange(
+            max((galaxy_catalog['ztrue']-10.*galaxy_catalog['pzsigma']).min(), 0.),
+            (galaxy_catalog['ztrue']+10.*galaxy_catalog['pzsigma']).max()+pz_bin_width,
+             pz_bin_width)
+        galaxy_catalog['pzpdf'] = gaussian(
+            galaxy_catalog.pzpdf_info['zbins'],
+            galaxy_catalog['ztrue'][:,None],
+            galaxy_catalog['pzsigma'][:,None])
+    elif galaxy_catalog.pzpdf_info['type']=='individual_bins':
+        galaxy_catalog['pzbins'] = [
+            np.arange(zmin, zmax, pz_bin_width)
+            for zmin, zmax in zip(
+                galaxy_catalog['ztrue']-10.*galaxy_catalog['pzsigma'],
+                galaxy_catalog['ztrue']+10.*galaxy_catalog['pzsigma'])]
+        galaxy_catalog['pzpdf'] = [gaussian(row['pzbins'], row['ztrue'], row['pzsigma'])
+                                   for row in galaxy_catalog]
```

Comment:
@m-aguena - Replacing `ztrue` by `z` in this section allows the recover the original behaviour from the `main` branch for the `demo_compute_deltasigma_weights` notebook (see my post on Slack for the figures). I think we need to discuss what implementation make more sense.

Reply:
@combet I centered the pdzs o `z` back again. I am just wondering if the figure of the weights on the notebook should be made as a function of `z` or `ztrue`

Reply:
Thanks @m-aguena - Showing the scattered version w.r.t `ztrue` allows us to highlight that two galaxies with the same redshift won't necessary pass a given 'probability cut' of being a background galaxy the same way. I think it has a bit more information than showing it w.r.t to `z` where, by construction of the mock data pdf, 2 galaxies with the same point-estimate redshift will behave the same way in that figure, as we gave them the same pdz. Is that making sense?

Reply:
@combet yes, I see your point. I decided to update the `demo_compute_deltasigma_weights` with more details.
The PR is ready to be reviewed now. Thanks!